source/grid/grid_generator.cc

```diff
@@ -28,11 +30,1180 @@
+                  const double l_y = D(1) - C(1);
+                  const double l_h = D(1) - l_y;
+                  const double by  = -l_h / length_b1_x * (node(0) - H(0));
+                  const double dy  = (height - by) / n_cells_y;
+                  const int    iy  = std::round((std::abs(node(1)) - by) / dy);
+                  const double dx  = length_b1_x / n_cells_x_1;
+                  const int    ix =
+                    std::round(std::abs(node(0) - center_mesh) / dx);
+
+                  const double alpha = bias_alpha(1 - (1.0 * iy) / n_cells_y);
+                  // define points on upper/lower horizontal far field side,
+                  // i.e. face DG or FI incline factor to move points G and I in
+                  // to the right by distance incline_facor*lenght_b2
+                  const Point<2> p(
+                    ix * dx + center_mesh +
+                      incline_factor * length_b2 * ix / n_cells_x_1,
+                    ((cell.material_id() == 2) ? (height) : (-height)));
+                  // interpolate between y = height and upper airfoil points
+                  // (block2) or y = -height and lower airfoil points (block5)
+                  node = airfoil_1D[((cell.material_id() == 2) ? (0) : (1))]
+                                   [n_cells_x_0 + ix] *
+                           alpha +
+                         p * (1 - alpha);
+                }
+              else if (cell.material_id() == 3 ||
+                       cell.material_id() == 6) // block 3 and 6
+                {
+                  // compute indices ix and iy
+                  const double dx = length_b2 / n_cells_x_2;
+                  const double dy = height / n_cells_y;
+                  const int    ix = std::round(std::abs(node(0) - H(0)) / dx);
+                  const int    iy = std::round(std::abs(node(1)) / dy);
+
+                  const double alpha_y = bias_alpha(1 - 1.0 * iy / n_cells_y);
+                  const double alpha_x =
+                    bias_alpha(1 - ((double)ix) / n_cells_x_2);
+                  // define on upper/lower horizontal far field side at y = +/-
+                  // height, i.e. face GK or IL incline factor to move points G
+                  // and H to the right
+                  const Point<2> p1(
+                    J(0) - (1 - incline_factor) * length_b2 * (alpha_x),
+                    ((cell.material_id() == 3) ? (height) : (-height)));
+                  // define points on HJ but use tail_y as y-coordinate, in case
+                  // last airfoil point has y =/= 0
+                  const Point<2> p2(J(0) - alpha_x * length_b2, tail_y);
+                  node = p1 * (1 - alpha_y) + p2 * alpha_y;
+                }
+              else
+                {
+                  Assert(false, ExcIndexRange(cell.material_id(), 1, 6));
+                }
+            }
+      }
+
+
+      /*
+       * This function returns a bias factor 'alpha' which is used to make the
```

Comment:
You could mention that this is bijective function mapping from [0,1] onto [0,1]. Are values near 0 or near 1 made tighter?